src/Sylius/Bundle/ApiBundle/Resources/config/routing/checkout.yml

```diff
@@ -15,8 +15,106 @@ sylius_api_checkout_index:
+        _sylius:
+            form:
+                type: sylius_checkout_address
+                options:
+                    customer: expr:service('sylius.repository.customer').find($customerId)
+            repository:
+                method: find
+                arguments: [$orderId]
+            state_machine:
+                graph: sylius_order_checkout
+                transition: address
+
+sylius_api_checkout_readdress:
+    path: /readdress/{orderId}
+    methods: [PUT]
+    defaults:
+        _controller: sylius.controller.order:applyStateMachineTransitionAction
+        _sylius:
+            repository:
+                method: find
+                arguments: [$orderId]
+            state_machine:
+                graph: sylius_order_checkout
+                transition: readdress
+
+sylius_api_checkout_select_shipping:
+    path: /select-shipping/{orderId}
+    methods: [PUT]
+    defaults:
+        _controller: sylius.controller.order:updateAction
+        _sylius:
+            form: sylius_checkout_select_shipping
+            repository:
+                method: find
+                arguments: $orderId
+            state_machine:
+                graph: sylius_order_checkout
+                transition: select_shipping
+
+sylius_api_checkout_reselect_shipping:
+    path: /reselect-shipping/{orderId}
+    methods: [PUT]
     defaults:
-        _controller: sylius.controller.checkout:proceedAction
+        _controller: sylius.controller.order:applyStateMachineTransitionAction
+        _sylius:
+            repository:
+                method: find
+                arguments: [$orderId]
+            state_machine:
+                graph: sylius_order_checkout
+                transition: reselect_shipping
+
+sylius_api_checkout_select_payment:
+    path: /select-payment/{orderId}
+    methods: [PUT]
+    defaults:
+        _controller: sylius.controller.order:updateAction
+        _sylius:
+            form: sylius_checkout_select_payment
+            repository:
+                method: find
+                arguments: $orderId
+            state_machine:
+                graph: sylius_order_checkout
+                transition: select_payment
+
+sylius_api_checkout_reselect_payment:
+    path: /reselect-payment/{orderId}
+    methods: [PUT]
+    defaults:
+        _controller: sylius.controller.order:applyStateMachineTransitionAction
+        _sylius:
+            repository:
+                method: find
+                arguments: [$orderId]
+            state_machine:
+                graph: sylius_order_checkout
+                transition: reselect_payment
+
+sylius_api_checkout_complete:
+    path: /complete/{orderId}
+    methods: [PUT]
+    defaults:
+        _controller: sylius.controller.order:updateAction
+        _sylius:
+            form: sylius_checkout_complete
+            repository:
+                method: find
+                arguments: $orderId
```

Comment:
To use a brackets or not to use them - that is the question!

Three out of seven arguments for find method does not have them ;)